### ISSUE TITLE
hi,every buddy,any help,thanks

### ISSUE THREAD

**Issue by mvwtest** [COM8/esp32-xmpp-iot#2]
I tested in ubuntu,debian,kali,parrots ,windows7/10...system,
And idf.py build the hello_world is no problem .olso the bin can run on esp chips  too.

but this project , idf.py build it, i get below problem from any systems:
----------------------------------------------------------------------------------------------------------------------------------

- Building ESP-IDF components for target esp32
CMake Error at /home/mvw/esp/esp-idf/tools/cmake/build.cmake:200 (message):
  Failed to resolve component 'libsodium'.
Call Stack (most recent call first):
  /home/mvw/esp/esp-idf/tools/cmake/build.cmake:228 (__build_resolve_and_add_req)
  /home/mvw/esp/esp-idf/tools/cmake/build.cmake:229 (__build_expand_requirements)
  /home/mvw/esp/esp-idf/tools/cmake/build.cmake:515 (__build_expand_requirements)
  /home/mvw/esp/esp-idf/tools/cmake/project.cmake:422 (idf_build_process)
  CMakeLists.txt:20 (project)

-- Configuring incomplete, errors occurred!
See also "/home/mvw/esp/esp32-xmpp-iot/ESP32/build/CMakeFiles/CMakeOutput.log".
cmake failed with exit code 1
----------------------------------------------------------------------------------------------------------------------------------
any help,thanks

**Comment by COM8** [comment]
You are missing libsodium.

On Debian / Ubuntu:

```shell
apt-get install libsodium-dev libmbedtls-dev
```

With Homebrew on MacOS:

```shell
brew install libsodium mbedtls
```

**Comment by mvwtest** [comment]
Thank you so much for your repley.
I  installed  libsodium-dev +  libmbedtls-dev.
and try rebuild ,but ,the surprise is non hug  me.
problem no any change,

ubuntu20.04     ESP-IDF4.3.2 maybe ?
if you are enjoying holiday. good luck for you . thank you again

**Comment by COM8** [comment]
Could you please provide me with the full build log since for me that solves the issue. I'm using `v4.3` of the ESP-IDF on Fedora 35.

**Comment by mvwtest** [comment]
thank you.
this is CMakeError.log:
------------------------------------------------------------------------------------------------------------------
Determining if the strtod_l exist failed with the following output:
Change Dir: /home/mvw/esp/esp32-xmpp-iot/ESP32/build/CMakeFiles/CMakeTmp

Run Build Command(s):/usr/bin/ninja cmTC_8b25b && [1/2] Building C object CMakeFiles/cmTC_8b25b.dir/CheckSymbolExists.c.obj
FAILED: CMakeFiles/cmTC_8b25b.dir/CheckSymbolExists.c.obj 
/home/mvw/.espressif/tools/xtensa-esp32-elf/esp-2021r2-8.4.0/xtensa-esp32-elf/bin/xtensa-esp32-elf-gcc    -o CMakeFiles/cmTC_8b25b.dir/CheckSymbolExists.c.obj   -c CheckSymbolExists.c
CheckSymbolExists.c: In function 'main':
CheckSymbolExists.c:8:19: error: 'strtod_l' undeclared (first use in this function); did you mean 'strtoull'?
   return ((int*)(&strtod_l))[argc];
                   ^~~~~~~~
                   strtoull
CheckSymbolExists.c:8:19: note: each undeclared identifier is reported only once for each function it appears in
ninja: build stopped: subcommand failed.


File /home/mvw/esp/esp32-xmpp-iot/ESP32/build/CMakeFiles/CMakeTmp/CheckSymbolExists.c:
/* */
#include <stdlib.h>

int main(int argc, char** argv)
{
  (void)argv;
#ifndef strtod_l
  return ((int*)(&strtod_l))[argc];
#else
  (void)argc;
  return 0;
#endif
}
------------------------------------------------------------------------------------------------------------------
and here is :CMakeOutput.log
[https://github.com/mvwtest/esp32-xmpp-iot/blob/master/CMakeOutput.log](url)

thank you.

**Comment by COM8** [comment]
Ah, sorry I meant the console log of running `idf.py build` since this is a problem of [Smooth](https://github.com/PerMalmberg/Smooth) I'm using as a dependency. Please make sure you run a `idf.py fullclean` before.

**Comment by mvwtest** [comment]
thank you,this issues is still love me.
now,i reinstall the Fedora35 and build up esp32 idf. try  again.good luck 4 me.

thrught  your project,i hope make a xmpp-Uart bridge,
use webpage login JID,and,TCP,Uart connect to arduino or other MCU.
for chidren ,student geeker,...i think,this is good gif.
 the esp32 vs RasPi,for cost, is a killer
but, i need deep learn it, thank you,
keep touch,

**Comment by COM8** [comment]
I don't think this is the right project for your use case, since this project is just a test for my own XMPP IoT protocol.
It is more or less a hard coded XMPP library written by me without any good security practices since I'm talking to the server over an unencrypted connection and it does support only plain authentication (sends the password in BASE64 play over the network).

What you can/should try is getting [libstrophe](https://strophe.im/libstrophe) to run on your ESP.
